### DISCUSSION TITLE
How to open file through the JSON IPC interface?

### DISCUSSION THREAD

**Discussion by TheOddler** [mpv-player/mpv#14743] [Q&A] [answered]
Hey!

I'm trying to run mpv using the `--input-ipc-server` option so I can control it with another program. This works fine, I can send commands like `cycle pause` and stuff, which works, but I haven't been able to figure out how to use that interface to set/change the currently playing file.

I tried sending something like `{ \"command\": [\"set_property\", \"external-file\", \"/home/user/Videos/video.mp4\"] }`, also tried `file`, `path`, and  `playlist` instead of `external-file`, but none work.

When running mpv with debug logging enabled and I send that command, I get this error:

```
[cplayer] Set property: external-file="/home/pablo/Videos/video.mp4" -> -3
```

What is the correct way of setting/changing the playing file through commands like this?

Thank you!

**Comment by guidocella** [comment, marked as answer]
https://mpv.io/manual/master/#command-loadfile

**Reply by TheOddler** [reply]
Thank you! I don't know how I missed that option, but just tested and works as expected.
